File: src/examples/cpp/signature_schema.cpp
#include "axllm/axllm.hpp"
#include <iostream>

int main() {
  axllm::Value signature = axllm::s("question:string -> answer:string");
  axllm::Value schema = axllm::to_json_schema(axllm::Core::get(signature, "outputs"));

  std::cout << axllm::stringify(schema) << "\n";
}

File: src/examples/go/signature_schema.go
package main

import (
	"fmt"

	ax "github.com/ax-llm/ax/go"
)

func main() {
	sig := ax.NewSignature("question:string -> answer:string")
	schema := sig.ToJSONSchema(nil).(map[string]ax.Value)
	fmt.Println("go-signature-schema", "outputs", len(sig.GetOutputFields()), "schema", schema["type"])
}
